automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInteractionSpec.scala

```diff
@@ -0,0 +1,162 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import java.io.File
+import java.time.Instant
+
+import org.broadinstitute.dsde.firecloud.api.Orchestration
+import org.broadinstitute.dsde.workbench.{ResourceFile, WebBrowserSpec}
+import org.broadinstitute.dsde.workbench.config.AuthToken
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.scalatest.time.{Seconds, Span}
+import org.scalatest.{BeforeAndAfterAll, FreeSpec}
+
+class NotebookInteractionSpec extends FreeSpec with WebBrowserSpec with LeonardoTestUtils with BeforeAndAfterAll {
+  implicit override val patienceConfig: PatienceConfig = PatienceConfig(timeout = scaled(Span(5, Seconds)))
+
+  /*
+   * This class creates a cluster in a new billing project and runs all tests inside the same cluster.
+   */
+  var billingProject : GoogleProject = _
+  var ronCluster : Cluster = _
+
+  implicit val ronToken: AuthToken = ronAuthToken
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+
+    billingProject = createNewBillingProject()
+    ronCluster = createNewCluster(billingProject)(ronAuthToken)
+  }
+
+  override def afterAll(): Unit = {
+    deleteAndMonitor(billingProject, ronCluster.clusterName)(ronAuthToken)
+    cleanupBillingProject(billingProject)
+
+    super.afterAll()
+  }
+
+  "Leonardo notebooks" - {
+
+    "should open the notebooks list page" in withWebDriver { implicit driver =>
+      withNotebooksListPage(ronCluster) { _ =>
+        // no-op; just verify that it opens
+      }
+    }
+
+    "should upload a notebook to Jupyter" in withWebDriver { implicit driver =>
+      val file = ResourceFile("diff-tests/import-hail.ipynb")
+        withNotebookUpload(ronCluster, file) { _ =>
+          // no-op; just verify that it uploads
+        }
+      }
+
+    val fileName: String = "import-hail.ipynb"
```

Comment:
I know this was there before, but maybe these `val`s could go inside ` "should verify notebook execution"` since I think they are specific to that test?

Reply:
It's actually used by a few so I'd prefer to move it up